src/Application/Actions/Sitemap.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MatchBot\Application\Actions;
+
+use JetBrains\PhpStorm\Pure;
+use MatchBot\Application\Environment;
+use MatchBot\Domain\CampaignRepository;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Psr\Log\LoggerInterface;
+use Safe\DateTimeImmutable;
+use SimpleXMLElement;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
+
+/**
+ * Creates a sitemap for the Big Give public facing site. See spec at https://www.sitemaps.org/protocol.html#prioritydef
+ */
+class Sitemap extends Action
+{
+    #[Pure]
+    public function __construct(
+        private CacheInterface $cache,
+        private CampaignRepository $campaignRepository,
+        private Environment $environment,
+        private \DateTimeImmutable $now,
+        LoggerInterface $logger,
+    ) {
+        parent::__construct($logger);
+    }
+
+    #[\Override]
+    protected function action(Request $request, Response $response, array $args): Response
+    {
+        // initial implementation of sitemap - following spec from https://www.sitemaps.org/protocol.html#prioritydef
+        // which is what Google uses.
+
+        // For now just listing up to 100k campaign and (if open) donate pages.
+
+        // @todo:
+        // - add charity pages as well as campaigns.
+        // - add metacampaigns
```

Comment:
To my mind this is probably the highest priority/ maybe the only essential item on this list.